### PR TITLE
[backend/frontend] refactor: use generic methods for all document types

### PR DESCRIPTION
**Pull request by Kroustille**
# Context
> Briefly describe the purpose and context of this PR. Include relevant screenshots or screen recordings for the product team.

This PR removes specific mutations for csv feeds, vault and scenarios. 
The same `useDocumentContext` hook is used to have the same behaviour for each document types.
It changes uses of old document methods in tests.
Behaviour to retrieve document type and metadata keys is refactored and fully tested.
There still some cleaning to do with an old document creation method, and csv feed, dashboards and scenarios queries but it will be done in a future PR.

Create, update and delete dashboards, csv feeds and scenarios.

## Related issues

- Related to #1471 
- Related to #1450 

# Checklist

## Quality

- [x] I consider this work complete and ready for review
- [x] I tested all features and edge cases
- [x] I refactored code where necessary to improve quality
- [x] I made sure my code meets development guidelines
- [x] I performed a self-review of my code

## Testing

- [x] I wrote test cases for the relevant use cases (unit, integration or e2e tests)
- [ ] (Bug fixes only) I added tests that reproduce and verify the fix

### What tests have been made

- [x] Unit tests
- [x] Integration tests
- [ ] E2E tests
- [x] Local manual tests

# Additional information
> Any additional context, dependencies, or concerns reviewers should know about.

### CODE REVIEW

File: apps/portal-api/src/modules/services/document/document.app.ts [hunk 1]
```diff
@@ -180,8 +206,26 @@ export const DocumentApp = {
...
+    const metadataKeys: string[] | undefined =
+      DocumentMetadataMappedByServiceIdentifier[serviceDefinition.identifier](
+        metadata
+      );
+
+    const missingMetadataKeys = metadataKeys.filter(
+      (key) => !metadata.some((meta) => meta.key === key)
+    );
+    if (missingMetadataKeys.length) {
+      logApp.warn(
```
Comment (carinelebas) [inline apps/portal-api/src/modules/services/document/document.app.ts]: Same question here and in other places: should it not be logApp.error?
